core/contracts/financial_templates/PricelessPositionManager.sol

```diff
@@ -11,149 +11,173 @@ import "../Finder.sol";
 
     // Unique identifier for DVM price feed ticker.
-    bytes32 priceIdentifer;
-
-    /**
-     * @dev Time that this contract expires.
-     */
+    bytes32 public priceIdentifer;
+    // Time that this contract expires.
     uint public expirationTimestamp;
-    /**
-     * @dev Time that has to elapse for a withdrawal request to be considered passed, if no liquidations occur.
-     */
+    // Time that has to elapse for a withdrawal request to be considered passed, if no liquidations occur.
     uint public withdrawalLiveness;
 
+    event Transfer(address indexed oldSponsor, address indexed newSponsor);
+    event Deposit(address indexed sponsor, uint indexed collateralAmount);
+    event Withdrawal(address indexed sponsor, uint indexed collateralAmount);
+    event RequestWithdrawal(address indexed sponsor, uint indexed collateralAmount);
+    event RequestWithdrawalExecuted(address indexed sponsor, uint indexed collateralAmount);
+    event RequestWithdrawalCanceled(address indexed sponsor, uint indexed collateralAmount);
+    event PositionCreated(address indexed sponsor, uint indexed collateralAmount, uint indexed tokenAmount);
+    event Redeem(address indexed sponsor, uint indexed collateralAmount, uint indexed tokenAmount);
+    event ContractExpired(address indexed caller);
+    event SettleExpiredPosition(
+        address indexed caller,
+        uint indexed collateralAmountReturned,
+        uint indexed tokensBurned
+    );
+
+    modifier onlyPreExpiration() {
+        require(getCurrentTime() < expirationTimestamp, "Cannot operate on a position past its expiry time");
+        _;
+    }
+
+    modifier onlyPostExpiration() {
+        require(getCurrentTime() >= expirationTimestamp, "Cannot operate on a position before its expiry time");
+        _;
+    }
+
+    modifier onlyCollateralizedPosition(address sponsor) {
+        require(positions[sponsor].collateral.rawValue > 0, "Position has no collateral and so is invalid");
+        _;
+    }
+
     constructor(
+        bool _isTest,
         uint _expirationTimestamp,
         uint _withdrawalLiveness,
         address _collateralAddress,
-        bool _isTest,
         address _finderAddress,
-        bytes32 _priceFeedIdentifier
+        bytes32 _priceFeedIdentifier,
+        string memory _syntheticName,
+        string memory _syntheticSymbol
     ) public Testable(_isTest) {
         expirationTimestamp = _expirationTimestamp;
         withdrawalLiveness = _withdrawalLiveness;
         collateralCurrency = IERC20(_collateralAddress);
-        // TODO: add parameters to register the synthetic token's name and symbol.
-        // TODO: add the collateral requirement. This is needed at settlement and at dispute resolution.
-        // TODO: validate the input of the finder/price identifier inputs.
-        Token mintableToken = new Token();
+        Token mintableToken = new Token(_syntheticName, _syntheticSymbol, 18);
```

Comment:
I see that this works, but it's very surprising to me that you don't need to inherit from ExpandedIERC20 in order to save mintableToken as an ExpandedIERC20 :/

Reply:
Ye, for sure. In fact I think we can remove `ExpandedIERC20` from all places in this contract and just use `Token` everywhere for the synthetic token. I have made this change and it works fine and I think it is more logical.